orderer/consensus/etcdraft/consenter.go

```diff
@@ -104,6 +106,53 @@ func (c *Consenter) ReceiverByChain(channelID string) MessageReceiver {
 	return nil
 }
 
+// ValidateJoinBlock implements the JoinBlockValidator interface and validates the etcdraft metadata
+// in the join block passed to the channel participation API. It performs checks to ensure the orderer
+// won't crash when calling the consenter's HandleChain.
+// It also determines whether the node is a member of the consenter set defined in the join block.
+func (c *Consenter) ValidateJoinBlock(configBlock *common.Block) (isClusterMember bool, err error) {
+	// Fetch consenter metadata if not nil
+	var metadata *common.Metadata
+	// Metadata can be nil if migration
+	if configBlock.Metadata != nil {
+		metadata, err = protoutil.GetConsenterMetadataFromBlock(configBlock)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to get consenter metadata from config block")
+		}
+	}
+
+	// Fetch etcdraft metadata
+	etcdRaftMetadata, err := ConsensusMetadataFromConfigBlock(configBlock)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get etcdraft metadata from config block")
+	}
+
+	// Validate etcdraft metadata
+	err = CheckConfigMetadata(etcdRaftMetadata)
+	if err != nil {
+		return false, err
+	}
+
+	// Check if member of consenters in join block
+	blockMetadata, err := ReadBlockMetadata(metadata, etcdRaftMetadata)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to read Raft metadata")
+	}
+
+	consenters := CreateConsentersMap(blockMetadata, etcdRaftMetadata)
+
+	_, err = c.detectSelfID(consenters)
```

Comment:
I think instead it's better to just initialize a  ConsenterCertificate  instance and invoke its IsConsenterOfChannel . 

All you need to know is if you're in the channel or not, you don't need the ID.

Reply:
I think the idea here was we wanted to mimic as closely as possible what's actually called in `HandleChain` since that's where things would potentially break if not valid. Because `HandleChain` calls `detectSelfID` we do something similar in the validate. I can see it reaching the same conclusion though so I'd be indifferent with whichever approach we want to use.

Reply:
Checking the certificate is byte-wise equal is a more strict check than what is done in the **detectSelfID**, so it's not possible that your earlier (byte-wise) check passes, but the new one fails.